### PR TITLE
Add button background animation and button disabler

### PR DESCRIPTION
**Pull request by jfranciscosousa**
Why:

* Button should be more noticable, and forms should not be submited without inputs

This change addresses the need by:

* Extracting the form to a another component, binding form inputs to state, validating that state (check if they are present), if this is true, allow the button to not be disabled and allow the default behaviour of the form submit.

### CODE REVIEW

File: frontend/components/Button/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  return (
+    <div styleName={rootStyles}>
+      <Component href={href} onClick={handleClick}>
+        <div styleName="content">
+          <Text fontFamily="mono" color="red" weight="bold">
+            #inscreve-te
+          </Text>
+        </div>
+
```
Comment (luistorres) [inline frontend/components/Button/index.js]: 🔫

File: frontend/components/Button/index.css [hunk 2]
```diff
@@ -2,14 +2,60 @@
 @import '../../styles/colors.css';
 @import '../../styles/layout.css';
 
+$button-offset: 4px;
+
 .root {
-  display: flex;
+  position: relative;
+
+  a,
```
Comment (luistorres) [inline frontend/components/Button/index.css]: You can give that element a class and avoid this, right?

Comment (jfranciscosousa) [reply frontend/components/Button/index.css]: Yeah

File: frontend/components/Button/index.js (hunk 1 above)
Comment (jfranciscosousa) [reply frontend/components/Button/index.js]: I LIKE

Comment (luistorres) [reply frontend/components/Button/index.js]: 🤷‍♂️

**Review by luistorres — APPROVED** [review]

File: frontend/components/Button/index.js (hunk 1 above)
Comment (jfranciscosousa) [reply frontend/components/Button/index.js]: Removed the space to please the greater good.

**Pull request merged**